lib/functions/ui.functions.sh

```diff
@@ -37,6 +45,10 @@ toggle_spin() {
                 done
             ) &
             spin_pid=$!
+            process_trap_add INT '_ui_trap_sig'
```

Comment:
Just a note: no need for the quotes here (it'll be interpreted as a string parameter regardless).